app/policies/wiki_policy.rb

```diff
@@ -5,4 +5,36 @@ def initialize(user, wiki)
     @user = user
     @wiki = wiki
   end
+
+  class Scope
+    attr_reader :user, :scope
+
+    def initialize(user, scope)
+      @user = user
+      @scope = scope
+    end
+
+    def resolve
+      wikis = []
+      if user.role == 'admin'
+        wikis = scope.all
+      elsif user.role == 'premium'
+        all_wikis = scope.all
+        all_wikis.each do |wiki|
+          if !wiki.private? || wiki.user == user || wiki.users.include?(user)
+            wikis << wiki
+          end
+        end
+      else
+        all_wikis = scope.all
+        wikis = []
+        all_wikis.each do |wiki|
+          if !wiki.private? || wiki.users.include?(user)
+            wikis << wiki
+          end
+        end
+      end
+      wikis
+    end
+  end
```

Comment:
ok great...so i want this file to look more like this one...https://github.com/sanjayypatel/wiki-pocket/blob/master/app/policies/wiki_policy.rb#L37 ... see how from a glance i can understand a lot more of what is going on in this example vs your's